### PR TITLE
Add openssf scorecard scan

### PR DESCRIPTION
**Pull request by huiwangredhat**
I found the [Scorecard](https://github.com/ossf/scorecard?tab=readme-ov-file#scorecard-github-action) provides the GitHub Action which is the easiest way to use it.
So I copied it and did a minor update for the action trigger.

### CODE REVIEW

File: .github/workflows/scorecard.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  push:
+    branches: [ "main" ]
+  pull_request:
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
```
Comment (jpower432) [inline .github/workflows/scorecard.yml:20]: I recommend adding an `if` statement here to check which repository it is running in to make sure it does not run on pushes to forks.

**Review by jpower432 — COMMENTED** [review]
Looks good. One suggestion.

File: .github/workflows/scorecard.yml (hunk 1 above)
Comment (huiwangredhat) [reply .github/workflows/scorecard.yml:20]: Good point! I added the `if` statement.

**Review by jpower432 — APPROVED** [review]
LGTM

**Pull request merged**